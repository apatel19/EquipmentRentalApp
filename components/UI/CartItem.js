import React from 'react';
import {View, Text, StyleSheet, TouchableOpacity, Platform} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const CartItem = props => {
  return (
    <View style={styles.cartItem}>
      <Text style={styles.itemData}>
        <Text style={styles.quantity}>{props.quantity}</Text>
        <Text style={styles.mainText}>{props.title}</Text>
      </Text>
      <View style={styles.itemData}>
        <Text style={styles.mainText}>${props.amount}</Text>
        {props.deletable && (
          <TouchableOpacity
            onPress={props.onRemove}
            style={styles.deleteButton}>
            <Icon
              name={Platform.OS === 'android' ? 'md-trash' : 'ios-trash'}
              size={23}
              color={'red'}
            />
          </TouchableOpacity>
        )}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  cartItem: {
    padding: 10,
    backgroundColor: 'white',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginHorizontal: 20,
  },
  itemData: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  quantity: {
    fontSize: 16,
    color: '#888',
  },
  mainText: {
    fontSize: 16,
  },
  deleteButton: {
    marginLeft: 20,
  },
});

export default CartItem;
