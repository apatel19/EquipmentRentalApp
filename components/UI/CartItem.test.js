import React from 'react';
import CartItem from './CartItem';
import renderer from 'react-test-renderer';

test('Card snapshot match test', () => {
  const tree = renderer.create(<CartItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
