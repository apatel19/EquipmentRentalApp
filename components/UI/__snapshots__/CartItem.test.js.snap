// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot match test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginHorizontal": 20,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#888",
          "fontSize": 16,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    />
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      $
    </Text>
  </View>
</View>
`;
